### PR TITLE
Remove libinih from codebase and add it as a dependency instead

### LINKED ISSUE #1811: Consider replacing src/ini.cpp with an external library
This is a single file of 297 lines. 
```
728 rmills@rmillsm1:~/gnu/github/exiv2/0.27-maintenance/conan-build $ wc ../src/ini*
     297    1097    8513 ../src/ini.cpp
729 rmills@rmillsm1:~/gnu/github/exiv2/0.27-maintenance/conan-build $ 
```
I found this somewhere on the internet and edited to put it into the Exiv2 namespace.  I added this as a single file because I thought it would be simpler than adding a new library dependency.

Both Dan and Christoph disagree with my decision. 

With conan, adding new dependent libraries is easier than it used to be. 

I only use conan with Visual Studio.  For other platforms (macOS, Linux, Unix, MinGW and Cygwin) I find it easier to use the platform package manager to install the dependencies (libz, expat and curl).  These are all "major players".   I don't even know if the ini parser can be installed by the platform package manager and/or if there is an conan recipe.  So, removing this little file, could result in lots of work.  For sure, I wouldn't like to reach the point that building Exiv2 demands conan because I've never got conan to work on Cygwin and/or MinGW. #1224.

### PR DESCRIPTION
**Pull request by kevinbackhouse**
Fixes: #1811 

`ini.cpp` and `ini.hpp` are copied from this repo: https://github.com/benhoyt/inih. Most platforms have a `libinih-dev` library that we can add as a dependency, so we can remove `ini.cpp` and `ini.hpp` from the exiv2 codebase.

On most platforms, it is easy to replace `ini.cpp` and `ini.hpp` with a dependency, but some platforms only have a `libinih` library and not a `libinih-devel` library, which is insufficient for building exiv2. On those platforms, I had to update the Actions workflows to build inih from source. This is the list of platforms where I had to do that:

1. Centos Stream
2. ~~MSYS2~~
3. Cygwin

On several other platforms, I had to upgrade to a newer version to get a sufficiently recent version of libinih. For example, I upgraded Ubuntu from 20.04 to 22.04.

### CODE REVIEW

**Comment by kmilos** [comment]
Et voila for MSYS2: https://github.com/msys2/MINGW-packages/pull/14781

**Comment by kevinbackhouse** [comment]
> Et voila for MSYS2: [msys2/MINGW-packages#14781](https://github.com/msys2/MINGW-packages/pull/14781)

Wow, that was fast! I was looking into it and only just found that repo about a minute ago!

**Comment by kevinbackhouse** [comment]
I'll convert this PR to draft until https://github.com/msys2/MINGW-packages/pull/14781 is ready.

The build failure in CIFuzz should resolve itself automatically after this PR is merged, because OSS-Fuzz uses the install script from our main branch: https://github.com/google/oss-fuzz/blob/582e69cebf7a970faf4ffb0c5492e497d860cb12/projects/exiv2/Dockerfile#L19

**Comment by kmilos** [comment]
@kevinbackhouse Merged. It might take a couple of hours/days until the binary package actually trickles down to the repo...

File: .github/workflows/on_PR_windows_matrix.yml
Comment (kmilos) [inline .github/workflows/on_PR_windows_matrix.yml:118]: git no longer needed for MSYS2 jobs

Comment (kevinbackhouse) [reply .github/workflows/on_PR_windows_matrix.yml:118]:
```suggestion
```

**Comment by kevinbackhouse** [comment]
@piponazo: do these cmake/conan changes look ok to you?

File: src/CMakeLists.txt
Comment (piponazo) [inline src/CMakeLists.txt:251]: (related to previous comment) If the library is required we should check in the file `cmake/findDependencies.cmake` if the library is not found and fail. Afterwards, here we could remove the conditional.

**Review by piponazo — CHANGES_REQUESTED** [review]
It seems to me that libinih is a strong dependency and we should be more strict in checking that the dependency is properly found.

Apart from that, all the changes look good to me 👏

File: cmake/findDependencies.cmake
Comment (kevinbackhouse) [reply cmake/findDependencies.cmake]: Good idea! This is done now.

**Review by piponazo — APPROVED** [review]
LGTM! Thanks ❤️

**Comment by kmilos** [comment]
FWIW: https://cygwin.com/pipermail/cygwin/2023-January/252780.html (couldn't find any other channel for such requests...)

**Comment by kmilos** [comment]
There is follow-up re Cygwin: https://cygwin.com/pipermail/cygwin/2023-January/252781.html

Anyone able to test?

**Comment by hassec** [comment]
@kevinbackhouse This seems to have broken the [nighlty release build](https://github.com/Exiv2/exiv2/actions/runs/3852674209/jobs/6565003830)

**Comment by kevinbackhouse** [comment]
> @kevinbackhouse This seems to have broken the [nighlty release build](https://github.com/Exiv2/exiv2/actions/runs/3852674209/jobs/6565003830)

@hassec: sorry about that! I'll look into it.

**Comment by eddiezato** [comment]
I build statically with msys2/ucrt64 to get one exe. Now `exiv2.exe` asks for `libINIReader.dll` and I can't solve it with `-linih` or `-linireader`. Any help, please?

**Comment by kmilos** [comment]
> I build statically with msys2/ucrt64 to get one exe. Now `exiv2.exe` asks for `libINIReader.dll`

Yep, the MSYS2 package did not ship the static library in the default build options... An update should hit the repo in a couple of days.

**Comment by kmilos** [comment]
> On `alpine`, if you install `inih-dev` without `inih-inireader-dev`, you get that error.

Granted, this was not envisaged when I wrote the module. I briefly checked the "major" distros these are usually packaged together...

**Comment by Ryanf55** [comment]
> inih_inireader_NOTFOUND

On alpine, they are separate installs, seems like a quick one-liner could fix it.

**Pull request merged**
